### PR TITLE
fixed java.lang.IndexOutOfBoundsException

### PR DESCRIPTION
**Pull request by tingliu9**
fixed java.lang.IndexOutOfBoundsException

### CODE REVIEW

File: ribbon-loadbalancer/src/main/java/com/netflix/loadbalancer/WeightedResponseTimeRule.java [hunk 1]
```diff
@@ -196,8 +196,10 @@ public Server choose(ILoadBalancer lb, Object key) {
                         n++;
                     }
                 }
+                if(serverCount > serverIndex){
```
Comment (msigne) [inline ribbon-loadbalancer/src/main/java/com/netflix/loadbalancer/WeightedResponseTimeRule.java]: Unit test?

Comment (tingliu9) [reply ribbon-loadbalancer/src/main/java/com/netflix/loadbalancer/WeightedResponseTimeRule.java]: I have Added a unit test

**Comment by Aloren** [comment]
Please check my [solution](https://github.com/Netflix/ribbon/pull/365/files#diff-b4783b1b379978aa5c9d788f353588ddL181) to this problem.
I think that when such situation is encountered (nodes were added or removed from eureka), `WeightedResponseTimeRule` should use `RoundRobinRule`, because `accumulatedWeights` were calculated for different nodes. So just retrying the process of choosing node will choose node based on the luckiness of the `randomWeight` value.

**Comment by tingliu9** [comment]
Yes, It is a good solution.

**Comment by qiangdavidliu** [comment]
Thanks @tingliu9 and @Aloren . Going to close this one in favour of #365.

**Pull request closed without merging**